aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -245,7 +245,11 @@ impl AptosVM {
     ) -> (VMStatus, TransactionOutputExt) {
         let mut session = self.0.new_session(resolver, SessionId::txn_meta(txn_data));
 
-        match TransactionStatus::from(error_code.clone()) {
+        match TransactionStatus::from(error_code.clone()).translate_invariant_violation(
```

Comment:
The logic change here is pretty ugly but I couldn't find a better way to clean this up. Would appreciate any suggestions. cc @gerben-stavenga @vgao1996